application/comit_node/src/btsieve/client.rs

```diff
@@ -228,6 +264,46 @@ mod ethereum {
                 poll_client.fetch_transactions(&query)
             })
         }
+
+        fn transaction_and_receipt_first_result(
+            &self,
+            query: &QueryId<Ethereum>,
+        ) -> Box<dyn Future<Item = TransactionAndReceipt, Error = Error> + Send> {
+            let poll_client = self.client.clone();
+            let query = query.clone();
+
+            poll_until_item(self.ethereum_poll_interval, move || {
+                let mut url = query.as_ref().clone();
+                url.set_query(Some("return_as=transaction_and_receipt"));
+
+                let transactions = poll_client
```

Comment:
:question: Isn't this variable name misleading if we're only returning the first result?

Reply:
This one will return all results so I would say the variable name is ok. The whole thing is in a closure that only returns the first item eventually.